### DISCUSSION TITLE
A simple summary of deploying Streampark

### DISCUSSION THREAD

**Discussion by fi-ecavc** [apache/streampark#3204] [General]
Let me briefly summarize the problems encountered, which are still in progress:
1. When starting streampark, pay attention to using sudo or this root user, otherwise there will be no silent error.
2. Regarding the construction of private image libraries, the official documentation is vague, ordinary docker registry can not be used at all, try to use image libraries that support namespaces, such as Alibaba Cloud ACR, harbor, etc
3. There is a problem with the official pod template format, which causes the task to fail to execute and report an error directly, and the format is obviously abnormal but can run.
4. Can push the image image to the image library, but can not pull from the image library to the k8s cluster, prompt the access permission error, resulting in the task has been in initiating, check the relevant credentials are normal, puzzled.
5. A small suggestion: provide clear and accurate templates and documentation, and be able to run through a demo application in 15 minutes. The 15-minute rule.
6. I feel that the log system seems to be not timely and accurate, and after cleaning and deleting the logs directory in the middle, there is basically no log output after operation.

**Comment by wolfboys** [comment]
Thanks for your feedback. We will carefully consider the feedback you provided. Currently, StreamPark  restructuring deployment flink job on k8s using the k8s-operator approach, which aims to reduce the complexity for users to use StreamPark.

Propsola: https://github.com/apache/incubator-streampark/issues/2879
Design documentation: https://www.craft.me/s/LrTTzYHqSOLYqY
